lightning_client.go

```diff
@@ -1463,6 +1511,155 @@ func (s *lightningClient) ChannelBackups(ctx context.Context) ([]byte, error) {
 	return resp.MultiChanBackup.MultiChanBackup, nil
 }
 
+// getChannelEventUpdate converts an lnrpc.ChannelEventUpdate to the higher
+// level ChannelEventUpdate.
+func getChannelEventUpdate(rpcChannelEventUpdate *lnrpc.ChannelEventUpdate) (
+	*ChannelEventUpdate, error) {
+
+	// getOutPoint is a helper go convert a hash and output index to
+	// a wire.OutPoint object.
+	getOutPoint := func(txId []byte, idx uint32) (*wire.OutPoint, error) {
+		hash, err := chainhash.NewHash(txId)
+		if err != nil {
+			return nil, err
+		}
+
+		return &wire.OutPoint{
+			Hash:  *hash,
+			Index: idx,
+		}, nil
+	}
+
+	result := &ChannelEventUpdate{}
+	update := rpcChannelEventUpdate.Channel
+	var err error
+
+	switch rpcChannelEventUpdate.Type {
+	case lnrpc.ChannelEventUpdate_PENDING_OPEN_CHANNEL:
+		result.UpdateType = PendingOpenChannelUpdate
+
+		channelPoint := update.(*lnrpc.ChannelEventUpdate_PendingOpenChannel).PendingOpenChannel
```

Comment:
nit: can use gRPC helper methods like `rpcChannelEventUpdate.GetPendingOpenChannel()` for this and below.

Reply:
Yeah, that's much nicer, done.